tests/stubtest_third_party.py

```diff
@@ -26,10 +26,21 @@ def run_stubtest(dist: Path) -> bool:
     with open(dist / "METADATA.toml") as f:
         metadata = dict(tomli.loads(f.read()))
 
-    if not run_stubtest_for(metadata, dist):
+    if not has_py3_stubs(dist):
         print(f"Skipping stubtest for {dist.name}\n\n")
         return True
 
+    apt_packages = metadata.get("stubtest_apt_dependencies", [])
+    if apt_packages:
+        try:
+            apt_cmd = ["apt", "install", *apt_packages]
```

Comment:
- This will be difficult to run locally, unless you're fine with running the entire script as root. That seems like a bad idea to me.
- If the package is already installed (`dpkg --get-selections | cut -f1 | grep '^package$'`), there's no need to install it again.
- If `apt` or `dpkg` doesn't exist, we should skip instead of failing.

Reply:
Maybe we should add a `--install-apt` command line flag, which we enable in CI. If set, we try to run `apt install`, if not set, we just print a hint.

Reply:
Maybe name it `--sudo-apt-install`, to make it clear that it will invoke `sudo` internally. That might not be what you expect a script to do when ran without sudo.

Reply:
The script as written doesn't invoke sudo. Should we do that? Although I think it's safer not to do it for now.

Reply:
Maybe it would be best to first `sudo apt install` all packages needed, and then run the stubtest script. This means we need a separate script, or a long `python3 -c` command, to parse the TOML files though.

Reply:
I like that we try to install the packages where we need them. This way, stubtest will only fail for the specific package that needs that dependency if there are install errors, not the whole stubtest run.

Reply:
Maybe we should add a new and minimal script that installs the apt packages and assumes it is ran as root, `sudo python3 tests/install_apt_packages.py $STUBS`. It should be as simple as possible, so that most of the work is done not as root.

Reply:
I'll do this in a separate PR, to see if that will be any cleaner after all.